`arocc` does not behave as expected during errors if `-E` (preprocess only) is set

test.c:
```c
#INCLUDE "invalid"
```

Built with `-Doptimize=ReleaseFast`, the preprocessed output is printed and a 0 exit code is returned:
```
> arocc -E test.c
.\test.c:1:1: error: invalid preprocessing directive
#INCLUDE "invalid"
^
1 error generated.
# 1 "test.c" 1
# 1 "<builtin>" 1
# 186 "<builtin>"
# 1 "<command line>" 1
# 1 "test.c" 2

> echo %errorlevel%
0
```

Built with `-Doptimize=Debug`, the preprocessed output is printed and a 1 exit code is returned:
```
> arocc -E test.c
.\test.c:1:1: error: invalid preprocessing directive
#INCLUDE "invalid"
^
1 error generated.
# 1 "test.c" 1
# 1 "<builtin>" 1
# 186 "<builtin>"
# 1 "<command line>" 1
# 1 "test.c" 2

> echo %errorlevel%
1
```

Expected behavior (any optimize mode):

```
> arocc -E test.c
.\test.c:1:1: error: invalid preprocessing directive
#INCLUDE "invalid"
^
1 error generated.

> echo %errorlevel%
1
```

Without `-E`, the expected behavior is present:

```
> arocc test.c
.\test.c:1:1: error: invalid preprocessing directive
#INCLUDE "invalid"
^
1 error generated.

> echo %errorlevel%
1
```

--- a/src/aro/Driver.zig
+++ b/src/aro/Driver.zig
@@ -565,6 +565,11 @@ fn processSource(
     if (d.only_preprocess) {
         d.comp.renderErrors();
 
+        if (d.comp.diagnostics.errors != 0) {
+            if (fast_exit) std.process.exit(1); // Not linking, no need for cleanup.
+            return;
+        }
+
         const file = if (d.output_name) |some|
             std.fs.cwd().createFile(some, .{}) catch |er|
                 return d.fatal("unable to create output file '{s}': {s}", .{ some, util.errorDescription(er) })

